keeper/msg_server.go

```diff
@@ -161,3 +160,27 @@ func (k *Keeper) SetAllowedDenoms(ctx context.Context, msg *types.MsgSetAllowedD
 		CurrentDenoms:  msg.Denoms,
 	})
 }
+
+// ValidateAccountFields validates if an account is eligible to be used as a forwarding account.
+//
+// A valid forwarding account must satisfy one of these conditions:
+//
+// 1. Is a new account:
+//   - A nil PubKey, i.e. the account never sent a transaction AND
+//   - A 0 sequence.
+//
+// 2. Is an account registered signerlessy:
+//   - A non nil PubKey with the custom type, i.e. the account has been registered sending a
+//     signerless tx.
+//   - Can have any sequence value.
+func ValidateAccountFields(account sdk.AccountI, address sdk.AccAddress) error {
+	pubKey := account.GetPubKey()
+
+	isNewAccount := pubKey == nil && account.GetSequence() == 0
+	isValidPubKey := pubKey != nil && account.GetPubKey().Equals(&types.ForwardingPubKey{Key: address})
```

Comment:
This can be simplified by reusing the shared variable!

```suggestion
	isValidPubKey := pubKey != nil && pubKey.Equals(&types.ForwardingPubKey{Key: address})
```